Topics/Tech_Stacks/JsonParsing.md

```diff
@@ -0,0 +1,231 @@
+
+ * String: Represents a sequence of characters enclosed in double quotation marks (")
+  
+    Example: "Hello, World!" 
+
+
+ * Number: Represents numeric values, including integers and floating-point numbers.
+    
+    Examples: 42, 3.14, -17 
+
+ * Boolean: Represents a logical value, either true or false. 
+    
+    Examples: true, false 
+
+ * Null: Represents an empty value or the absence of a value. 
+    
+    Example: null
+
+These primitive data types can be used alone or combined to create more complex JSON structures such as objects and arrays. For example, an object may contain key-value pairs where the values can be strings, numbers, booleans, null, or even nested objects and arrays.
+
+### Complex Data Types
+
+JSON allows for the construction of more complex data structures beyond primitive data types by using objects and arrays. 
+
+ * Objects: An object in JSON is an unordered collection of key-value pairs. Key-value pairs are separated by commas and enclosed in curly braces {}. Keys must be strings, and values can be strings, numbers, booleans, null, objects, or arrays. 
+ Example:
+
+
+{
+  "name": "John Doe",
+  "age": 30,
+  "isStudent": false,
+  "address": {
+    "city": "Exampleville",
+    "zipcode": "12345"
+  }
+}
+
+ * Array: An array in JSON is an ordered list of values. Values are separated by commas and enclosed in square brackets []. Values can be strings, numbers, booleans, null, objects, or other arrays. 
+ Example:
+
+ [
+  "apple",
+  "banana",
+  "orange",
+  {
+    "color": "red",
+    "quantity": 5
+  }
+]
+
+## JSON Parsing in Different Programming Languages
+
+### Python Parse JSON
+
+#### Parse JSON String in Python
+
+Python has a built in module that allows you to work with JSON data. At the top of your file, you will need to import the json module.
+
+```{python}
+import json
+```
+
+
+If you need to parse a JSON string that returns a dictionary, then you can use the json.loads() method.
+```{python}
+import json
+
+# assigns a JSON string to a variable called jess 
+jess = '{"name": "Jessica Wilkins", "hobbies": ["music", "watching TV", "hanging out with friends"]}'
+
+# parses the data and assigns it to a variable called jess_dict
+jess_dict = json.loads(jess)
+
+# Printed output: {"name": "Jessica Wilkins", "hobbies": ["music", "watching TV", "hanging out with friends"]}
+print(jess_dict)
+
+```
+
+#### Parse and Read JSON File in Python
+
+Suppose we have a JSON file called fcc.json. If we want to read that file, we first need to use Python's built-in `open()` function with the mode of read. We are using the `with` keyword to make sure that the file is properly closed.
+
+```{python}
+with open('fcc.json', 'r') as fcc_file:
+```
+We can then parse the file using the json.load() method and assign it to a variable called fcc_data.
+
+```{python}
+fcc_data = json.load(fcc_file)
+```
+The final step would be to print the results.
+
+```{python}
+print(fcc_data)
+```
+
```

Comment:
Please include an entire code block for consistency, which include all the procedure above for user copy convenience.

Reply:
thank you for point it out! 
Added.